src/helpers/rectangle.ts

```diff
@@ -0,0 +1,28 @@
+import { Rect, Zone } from "../types";
+import { intersection } from "./zones";
+
+/**
+ * Compute the intersection of two rectangles. Returns nothing if the two rectangles don't overlap
+ */
+export function rectIntersection(rect1: Rect, rect2: Rect): Rect | undefined {
+  return zoneToRect(intersection(rectToZone(rect1), rectToZone(rect2)));
```

Comment:
It is useful to introduce two helpers to convert a rect to a non-really zone just to use `intersection` ? It's two `if` (overlap) and basic operations (math.Max/Min).
For now, `zoneToRect` and `rectToZone` are private, but could be easily exported. If it's the case, I could already see myself in 6 months trying to figure why the f* we convert a rect to a zone without the getters.
The only interest I see is that `intersection` is already tested.
What do you think ?